### PR TITLE
TDL-16888 Added code comments

### PR DESCRIPTION
**Pull request by prijendev**
# Description of change
- Add code comments for better readability

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_bing_ads/__init__.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ class InvalidDateRangeEnd(Exception):
     pass
 
 def log_service_call(service_method, account_id):
+    # Log the call of service
```
Comment (dbshah1212) [inline tap_bing_ads/__init__.py:64]: Instead of this add what this method doing

File: tap_bing_ads/__init__.py [hunk 2]
```diff
@@ -87,17 +88,22 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 class CustomServiceClient(ServiceClient):
+    # Provides an class for calling the methods of the specified Bing Ads service.
```
Comment (dbshah1212) [inline tap_bing_ads/__init__.py:91]:
```suggestion
    # This class calling the methods of the specified Bing Ads service.
```

File: tap_bing_ads/__init__.py (hunk 1 above)
Comment (prijendev) [reply tap_bing_ads/__init__.py]: Updated comment.

File: tap_bing_ads/__init__.py [hunk 3]
```diff
@@ -87,17 +88,22 @@ def wrapper(*args, **kwargs):
...
         return super().__init__(name, 'v13', **kwargs)
 
     def __getattr__(self, name):
+        # Set authorization data and options before every service call.
         service_method = super(CustomServiceClient, self).__getattr__(name)
         return log_service_call(service_method, self._authorization_data.account_id)
 
     def set_options(self, **kwargs):
         self._options = kwargs
+        # Ensemble the request header send to API services.
```
[the remark this answers is not in the document]
Comment (prijendev) [reply tap_bing_ads/__init__.py]: Updated.

File: tap_bing_ads/__init__.py [hunk 4]
```diff
@@ -87,33 +90,43 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 class CustomServiceClient(ServiceClient):
+    # This class calling the methods of the specified Bing Ads service.
     def __init__(self, name, **kwargs):
+        # Initializes a new instance of this ServiceClient class.
         return super().__init__(name, 'v13', **kwargs)
 
     def __getattr__(self, name):
+        # Log and return serivcecall(suds client call) object
```
Comment (savan-chovatiya) [inline tap_bing_ads/__init__.py:99]:
```suggestion
        # Log and return service call(suds client call) object
```

Comment (prijendev) [reply tap_bing_ads/__init__.py]: Updated

**Review by dbshah1212 — APPROVED** [review]

File: tap_bing_ads/__init__.py [hunk 5]
```diff
@@ -61,9 +61,11 @@ class InvalidDateRangeEnd(Exception):
     pass
```
Comment (karanpanchal-crest) [inline tap_bing_ads/__init__.py:61]: @dbshah1212 We also have `exclusions.py` and `reports.py` in this tap. Can you add comments in those files as well?

**Review by karanpanchal-crest — CHANGES_REQUESTED** [review]

File: tap_bing_ads/__init__.py (hunk 5 above)
Comment (prijendev) [reply tap_bing_ads/__init__.py:61]: @karanpanchal-crest `exclusions.py` and `reports.py` contain only constant with comments and there is no logical code defined in this.

**Review by karanpanchal-crest — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Pull request merged**